src/main/java/com/openshift/internal/restclient/model/Secret.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package com.openshift.internal.restclient.model;
+
+import static com.openshift.internal.restclient.capability.CapabilityInitializer.initializeCapabilities;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.jboss.dmr.ModelNode;
+
+import com.openshift.restclient.IClient;
+import com.openshift.restclient.model.secret.ISecret;
+import com.openshift.restclient.model.secret.SecretType;
+import com.openshift.restclient.utils.Base64Coder;
+
+/**
+ * @author Jiri Pechanec
+ */
+public class Secret extends KubernetesResource implements ISecret {
+
+	public Secret(ModelNode node, IClient client, Map<String, String []> propertyKeys) {
+		super(node, client, propertyKeys);
+		initializeCapabilities(getModifiableCapabilities(), this, getClient());
```

Comment:
This is not required unless you are initializing specific capabilites that apply to a Secret kind